packages/libs/src/sdk/services/AntiAbuseOracleSelector/AntiAbuseOracleSelector.ts

```diff
@@ -0,0 +1,57 @@
+import { sampleSize } from 'lodash'
+import { mergeConfigWithDefaults } from '../../utils/mergeConfigs'
+import type { LoggerService } from '../Logger'
+import { defaultAntiAbuseOracleSelectorConfig } from './constants'
+import type {
+  AntiAbuseOracleSelectorService,
+  AntiAbuseOracle,
+  AntiAbuseOracleSelectorConfig
+} from './types'
+
+export class AntiAbuseOracleSelector implements AntiAbuseOracleSelectorService {
+  private services: AntiAbuseOracle[] | null = null
+  private readonly logger: LoggerService
+  private readonly endpoints: string[]
+  private readonly addresses: string[]
+  constructor(config?: AntiAbuseOracleSelectorConfig) {
+    const configWithDefaults = mergeConfigWithDefaults(
+      config,
+      defaultAntiAbuseOracleSelectorConfig
+    )
+    this.endpoints = configWithDefaults.endpoints
+    this.addresses = configWithDefaults.addresses
+    this.logger = configWithDefaults.logger
+  }
+
+  public async getSelectedService() {
+    const services = await this.getHealthyRegisteredServices()
```

Comment:
do we always want this method to healthcheck everything if we're only picking one? maybe better to just pick and then check

Reply:
If we pick one and then check, then we're doing the requests to health check serially. Right now there are 2 unhealthy and 1 healthy, meaning 2/3 of the time we'll do at least 2 requests in series, and of that, half of the time we'll do 3 requests in series. Checking all at once we get the healthy service in one request iteration.

We could, though, have it biased towards using the healthy one first. But with three total services I don't think this is so bad.